core/predicate_check.go

```diff
@@ -35,6 +38,16 @@ func CheckPredicates(rules params.Rules, predicateContext *precompileconfig.Pred
 	// Prepare the predicate storage slots from the transaction's access list
 	predicateArguments := predicateutils.PreparePredicateStorageSlots(rules, tx.AccessList())
 
+	// If there are no predicates to verify, we must terminate early because the
+	// proposervm block context isn't guaranteed to be populated.
+	if len(predicateArguments) == 0 {
+		return predicateResults, nil
+	}
```

Comment:
```suggestion
	// If there are no predicates to verify, return early and skip requiring the proposervm block
	// context to be populated.
```

saying we must terminate early is a bit confusing to me here, I think it's better to explicitly call out the check we're skipping by returning early here.